src/main/java/co/rsk/federate/btcreleaseclient/cache/PegoutSignedCacheImpl.java

```diff
@@ -0,0 +1,47 @@
+package co.rsk.federate.btcreleaseclient.cache;
+
+import co.rsk.crypto.Keccak256;
+import com.google.common.annotations.VisibleForTesting;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+
+class PegoutSignedCacheImpl implements PegoutSignedCache {
+
+  private final Map<Keccak256, Instant> cache = new ConcurrentHashMap<>();
+  private final Duration ttl;
+
+  PegoutSignedCacheImpl(Duration ttl) {
+    this.ttl = ttl;
+  }
+
+  @Override
+  public boolean hasAlreadyBeenSigned(Keccak256 pegoutCreationRskTxHash) {
+    Instant currentTimestamp = Instant.now();
+
+    return Optional.ofNullable(pegoutCreationRskTxHash)
+        .map(cache::get)
+        .map(timestamp -> isValidTimestamp(currentTimestamp, timestamp))
+        .orElse(false);
+  }
+
+  @Override
+  public void put(Keccak256 pegoutCreationRskTxHash) {
+    Instant currentTimestamp = Instant.now();
+
+    Optional.ofNullable(pegoutCreationRskTxHash)
+        .ifPresent(rskTxHash -> cache.putIfAbsent(rskTxHash, currentTimestamp));
+  }
+
+  private boolean isValidTimestamp(Instant currentTimestamp, Instant timestamp) {
+    return currentTimestamp != null && timestamp != null &&
+        currentTimestamp.toEpochMilli() - timestamp.toEpochMilli() <= ttl.toMillis();
+  }
+
+  @VisibleForTesting
+  Map<Keccak256, Instant> getCache() {
```

Comment:
Left some comments in the test class. This probably isn't needed. We could stick to testing the class with the exposed methods. Or is it important to access the internal state of the cache? Maybe it's needed to check the timestamps used, let me know your thoughts

Reply:
Why not just add one more method to know the size of the cache? With this method, I think we should be able to test the class properly. Also, I consider this method useful to avoid checking if a list of peg-outs is in the cache when the cache is empty.